File: WoofIOS/SwiftPackages/Presentation/Grid/Sources/PresentationGrid/UI/Components/DogCell.swift
import DesignSystem
import PresentationCommon
import Resources
import SwiftUI

struct DogCell: View {
    let cell: DogCellUIModel
    let onCellAppear: (DogCellUIModel) -> Void
    let onCellTap: (DogCellUIModel) -> Void

    var body: some View {
        AsyncImageStateProvider(url: cell.photoUrl) { imageState in
            OrgPhotoCard(
                title: cell.name,
                imageState: imageState,
                contentDescription: L10n.Accessibility.dogPhoto(cell.name),
                onTap: { onCellTap(cell) }
            )
        }
        .onAppear {
            onCellAppear(cell)
        }
    }
}

File: WoofAndroid/presentation/grid/src/main/kotlin/com/perrystreet/woof/presentation/grid/ui/components/DogCell.kt
package com.perrystreet.woof.presentation.grid.ui.components

import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.ui.res.stringResource
import com.perrystreet.woof.designsystem.atomic.organisms.card.OrgPhotoCard
import com.perrystreet.woof.presentation.common.image.AsyncImageStateExtensions.rememberAsyncImageState
import com.perrystreet.woof.presentation.grid.uimodel.DogCellUIModel
import com.perrystreet.woof.resources.R

@Composable
internal fun DogCell(
    cell: DogCellUIModel,
    onCellAppear: (DogCellUIModel) -> Unit,
    onCellTap: (DogCellUIModel) -> Unit,
) {
    LaunchedEffect(cell.id) {
        onCellAppear(cell)
    }

    OrgPhotoCard(
        title = cell.name,
        imageState = rememberAsyncImageState(url = cell.photoUrl),
        contentDescription = stringResource(R.string.accessibility_dog_photo, cell.name),
        onTap = { onCellTap(cell) },
    )
}
